### PR TITLE
feat: add selectPortfolioCryptoHumanBalancesByFilter

### PR DESCRIPTION
**Pull request by DaoDev44**
## Description

- Add `selectPortfolioCryptoHumanBalancesByFilter` to show human readable crypto balance by assetId or assetId and accountId. AssetId is also required now for `selectPortfolioFiatBalancesByFilter`.
- Fix issue where assetBalances did not aggregate balances across multiple accounts.

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [x] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

If applicable, please link to the github issue and put `closes #XXXX` in your comment to auto-close the issue that your PR fixes.

## Testing

Please outline all testing steps

1. Pull branch locally and run `yarn` to install new deps
2. etc...

### CODE REVIEW

**Review by reallybeard — APPROVED** [review]

File: src/state/slices/portfolioSlice/portfolioSlice.ts [hunk 1]
```diff
@@ -388,6 +415,24 @@ export const selectPortfolioCryptoBalanceByAssetId = createSelector(
...
+  selectAssetIdParamFromFilter,
+  (assets, accountBalances, assetBalances, accountId, assetId): string => {
+    if (accountId && assetId) {
+      return fromBaseUnit(
+        bnOrZero(accountBalances[accountId][assetId]),
+        assets[assetId].precision ?? 0
+      )
+    }
+
+    return fromBaseUnit(bnOrZero(assetBalances[assetId]), assets[assetId].precision ?? 0)
```
Comment (0xdef1cafe) [inline src/state/slices/portfolioSlice/portfolioSlice.ts:432]: this should probably guard for `if (assetId) return assetBalances[assetId]...` and return `'0'` if it's not provided

Comment (0xdef1cafe) [reply src/state/slices/portfolioSlice/portfolioSlice.ts:432]: sorry, i'm wrong, looks like `assetId` is now a required part of the filter, this is good.

File: src/state/slices/portfolioSlice/portfolioSlice.ts [hunk 2]
```diff
@@ -388,6 +415,24 @@ export const selectPortfolioCryptoBalanceByAssetId = createSelector(
   (byId, assetId): string => byId[assetId]
 )
 
+export const selectPortfolioCryptoHumanBalancesByFilter = createSelector(
```
[the remark this answers is not in the document]
Comment (0xdef1cafe) [reply src/state/slices/portfolioSlice/portfolioSlice.ts:418]: slightly misleading pluralization - i'd expect this to return an object of balances or something

File: src/state/slices/portfolioSlice/portfolioSlice.ts [hunk 3]
```diff
@@ -161,8 +165,19 @@ export const accountToPortfolio: AccountToPortfolio = args => {
         portfolio.accounts.byId[accountSpecifier].push(caip19)
         portfolio.accounts.ids.push(accountSpecifier)
 
-        portfolio.assetBalances.byId[caip19] = ethAccount.balance
-        portfolio.assetBalances.ids.push(caip19)
+        // if asset already exist inside assetBalances, add balance to existing balance
+        if (!portfolio.assetBalances.byId[caip19]) {
+          portfolio.assetBalances.byId[caip19] = ethAccount.balance
+        } else {
+          portfolio.assetBalances.byId[caip19] = sumBalance(
```
Comment (0xdef1cafe) [inline src/state/slices/portfolioSlice/portfolioSlice.ts:172]: we can simply this to just the else clause and always sum it by doing `sumBalance(portfolio.assetBalances.byId[caip19] ?? '0', ethAccount.balance`

File: src/state/slices/portfolioSlice/portfolioSlice.ts [hunk 4]
```diff
@@ -161,8 +165,19 @@ export const accountToPortfolio: AccountToPortfolio = args => {
...
+            portfolio.assetBalances.byId[caip19],
+            ethAccount.balance
+          )
+        }
+
+        if (!portfolio.assetBalances.ids.includes(caip19)) {
```
Comment (0xdef1cafe) [inline src/state/slices/portfolioSlice/portfolioSlice.ts:178]: we can also do the `Array.from(new Set([...oldArray, newValue]))` pattern here to one line it, but i'm just being stylistic/nit picking here

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]
1 required change (pluralization), 1 or 2 optional simplifications, just a few other comments

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**